src/test/java/com/airlines/dao/impl/CrewDaoImplTest.java

```diff
@@ -0,0 +1,182 @@
+    private static final String CREATE_TABLES_QUERY = JdbcUtil.getSqlQueryString(SCHEMA_SQL_PATH);
+    private static final String GENERATE_DATA_QUERY = JdbcUtil.getSqlQueryString(DATA_SQL_PATH);
+
+    private static final CrewMember CREW_MEMBER = buildTestCrewMember();
+    private static final Crew CREW = buildCrewFromDataSet();
+    private static final Crew INVALID_CREW = buildInvalidCrewFromDataSet();
+
+    private static CrewDao crewDao;
+
+    private static DataSource dataSource;
+
+    @BeforeClass
+    public static void init() {
+        dataSource = TestDatasourceProvider.createTestDatasource();
+        crewDao = new CrewDaoImpl(dataSource);
+    }
+
+    @Before
+    public void dataBaseInit() throws SQLException {
+        createTables();
+    }
+
+    @Test
+    public void shouldThrowExceptionIfInvalidArgumentCrewMemberProvidedToAddMethod() {
+        IllegalArgumentException exception = assertThrows(IllegalArgumentException.class, () -> crewDao.add(null, null));
+        assertEquals("Valid crewMember entity should be provided", exception.getMessage());
+    }
+
+    @Test
+    public void shouldThrowExceptionIfInvalidArgumentCrewProvidedToAddMethod() {
+        IllegalArgumentException exception = assertThrows(IllegalArgumentException.class, () -> crewDao.add(CREW_MEMBER, null));
+        assertEquals("Valid crew entity should be provided", exception.getMessage());
+    }
+
+    @Test
+    public void shouldCorrectlyAddCrewMemberToCrew() throws SQLException {
+        populateTestData();
+        crewDao.add(CREW_MEMBER, CREW);
+        List<CrewMember> list = crewDao.getByCrewId(3L);
+        assertTrue(list.contains(CREW_MEMBER));
+    }
+
+    @Test
+    public void shouldGetCrewMembersByCrewId() throws SQLException {
+        populateTestData();
+        List<CrewMember> list = crewDao.getByCrewId(3L);
+        assertEquals(1, list.size());
+    }
+
+    @Test
+    public void shouldThrowExceptionIfCrewIsNotProvidedForGetByIdMethod() throws SQLException {
+        Crew invalidCrew = buildInvalidCrewFromDataSet();
+        IllegalArgumentException exception = assertThrows(IllegalArgumentException.class, () -> crewDao.getByCrewId(invalidCrew.getId()));
+        assertEquals("Cannot find crew members for crew without id", exception.getMessage());
+    }
+
+    @Test
+    public void getByCrewName() throws SQLException {
+        populateTestData();
+        List<CrewMember> list = crewDao.getByCrewName("Vityaz");
+        assertEquals(3, list.size());
+    }
+
+    @Test
+    public void shouldRemoveCrewMemberFromCrew() throws SQLException {
+        populateTestData();
+        crewDao.add(CREW_MEMBER, CREW);
+        List<CrewMember> list = crewDao.getByCrewId(3L);
+        assertTrue(list.contains(CREW_MEMBER));
+        crewDao.removeCrewMemberFromCrew(CREW, CREW_MEMBER);
+        List<CrewMember> listAfterRemove = crewDao.getByCrewId(3L);
```

Comment:
if there are more than 3 lines in test method, use given - when - then patter.
**given** (we have at the start - some preparatiob for test)
```
populateTestData();
crewDao.add(CREW_MEMBER, CREW);
```

**when** (we make an action, for example call to out method under test) we are doing
`List<CrewMember> list = crewDao.getByCrewId(3L);`

**then** (we receive some result, so we are checking it if it's correct)
assert
assert

Someone even writing it in the code like:
//when
...
//then
..
but you could just make one empty line between blocks (if there are 3-4 lines or more, if 2-3 it's ok)


Also in this method I susggest using or populateTestData, then you don't need to add member, just call remove:
```
        populateTestData();

        crewDao.removeCrewMemberFromCrew(CREW, CREW_MEMBER);

        List<CrewMember> listAfterRemove = crewDao.getByCrewId(3L);
        assertEquals(0, listAfterRemove.size());
        assertFalse(listAfterRemove.contains(CREW_MEMBER))
```
Or use only ADD method and not use populate, because I don't think you need both of them (populate data and add):
add crew member,
add crew
add crews_crew_members row
but it's sounds like an overhead.

Or as you mentioned kaizen,  it's a violation of **muda** principle
so let the first variant here, smth like:
```     
        populateTestData();
        List<CrewMember> initial = crewDao.getByCrewId(1L);

        crewDao.removeCrewMemberFromCrew(new Crew(1L, "Vityaz", new ArrayList<>()), CREW_MEMBER);

        assertTrue(initial.contains(CREW_MEMBER));
        assertEquals(3, initial.size());
        
        List<CrewMember> listAfterRemove = crewDao.getByCrewId(1L);
        assertEquals(2, listAfterRemove.size());
        assertFalse(listAfterRemove.contains(CREW_MEMBER));
 ```